File: day_6/task_1.py
def countOrbits(orbitsMap, a):
  return 0 if a not in orbitsMap else 1 + countOrbits(orbitsMap, orbitsMap[a])

with open('input.txt') as f:
  orbitsMap = {}
  for line in f:
    a, b = line.strip().split(')')
    orbitsMap[b] = a

  orbits = [countOrbits(orbitsMap, a) for a in orbitsMap]
  print(sum(orbits))

File: src/days/day1/task1.ts
import esMain from "es-main";
import { readFile } from "fs/promises";
import * as R from "ramda";

export const parseInput = async (path: string) => {
  const data = await readFile(new URL(path, import.meta.url), "utf-8");
  return data
    .split("\n\n")
    .map((inventory) => inventory.split("\n").map(Number));
};

const calcMostCalories = (inventories: number[][]) => {
  return R.sum(inventories.reduce(R.maxBy(R.sum)));
};

if (esMain(import.meta)) {
  const testInput = await parseInput("./test.txt");
  console.log("TEST:", calcMostCalories(testInput));

  const input = await parseInput("./input.txt");
  console.log("INPUT:", calcMostCalories(input));
}
